Content.Server/_Goobstation/Devil/Contract/DevilContractSystem.cs

```diff
@@ -0,0 +1,506 @@
+    }
+
+    private void UpdateContractWeight(Entity<DevilContractComponent> contract, PaperComponent? paper = null)
+    {
+        if (!Resolve(contract, ref paper))
+            return;
+
+        contract.Comp.CurrentClauses.Clear();
+        var newWeight = 0;
+
+        var matches = _clauseRegex.Matches(paper.Content);
+        foreach (Match match in matches)
+        {
+            if (!match.Success)
+                continue;
+
+            var clauseKey = match.Groups["clause"].Value.Trim().ToLowerInvariant().Replace(" ", "");
+
+            if (!_prototypeManager.TryIndex(clauseKey, out DevilClausePrototype? clauseProto)
+                || !contract.Comp.CurrentClauses.Add(clauseProto))
+                continue;
+
+            newWeight += clauseProto.ClauseWeight;
+        }
+
+        contract.Comp.ContractWeight = newWeight;
+    }
+
+    private void DoContractEffects(Entity<DevilContractComponent> contract, PaperComponent? paper = null)
+    {
+        if (!Resolve(contract, ref paper))
+            return;
+
+        UpdateContractWeight(contract);
+
+        var matches = _clauseRegex.Matches(paper.Content);
+        var processedClauses = new HashSet<string>();
+
+        foreach (Match match in matches)
+        {
+            if (!match.Success)
+                continue;
+
+            var targetKey = match.Groups["target"].Value.Trim().ToLowerInvariant().Replace(" ", "");
+            var clauseKey = match.Groups["clause"].Value.Trim().ToLowerInvariant().Replace(" ", "");
+
+            var locId = _targetResolvers.Keys.FirstOrDefault(id => Loc.GetString(id).Equals(targetKey, StringComparison.OrdinalIgnoreCase));
+            var resolver = _targetResolvers[locId];
+
+            if (resolver(contract.Comp) == null)
+            {
+                _sawmill.Warning($"Unknown resolver: {resolver(contract.Comp)}");
+                continue;
+            }
+
+            if (!_prototypeManager.TryIndex(clauseKey, out DevilClausePrototype? clause))
+            {
+                _sawmill.Warning($"Unknown contract clause: {clauseKey}");
+                continue;
+            }
+
+            // no duplicates
+            if (!processedClauses.Add(clauseKey))
+            {
+                _sawmill.Warning($"Attempted to apply duplicate clause: {clauseKey} on contract {ToPrettyString(contract)}");
+                continue;
+            }
+
+
+            var targetEntity = resolver(contract.Comp);
+
+            if (targetEntity is not null)
+                ApplyEffectToTarget(targetEntity.Value, clause, contract);
+            else
+                _sawmill.Warning($"Invalid target entity from resolver for clause {clauseKey} in contract {ToPrettyString(contract)}");
+        }
+    }
+
+    private void ApplyEffectToTarget(EntityUid target, DevilClausePrototype clause, Entity<DevilContractComponent>? contract)
+    {
+        //_sawmill.Debug($"Applying {clause.ID} effect to {ToPrettyString(target)}");
```

Comment:
Can remove this and the other commented-out debug logs below; or un-comment them